src/ui/game/userlist/user_cell.cpp

```diff
@@ -75,6 +76,30 @@ bool GlobedUserCell::init(const PlayerStore::Entry& entry, const PlayerAccountDa
         .id("percentage-label"_spr)
         .store(percentageLabel);
 
+    auto last_letter = typeinfo_cast<CCNode*>(nameLabel->getChildren()->objectAtIndex(data.name.length() - 1));
+
+    // No check function because of percentagelabel being moved
+    if (nameColor == ccc3(15, 239, 195)) {
+        createBadge(createLayout("role-mod.png"_spr, 1.f, last_letter->convertToWorldSpace(getPosition()) + CCPoint{20.f, 6.f}, "globed-mod-badge", nullptr)).parent(this);
+        percentageLabel->setPosition({percentageLabel->getPositionX() + 16.f, percentageLabel->getPositionY() - 6.f});
+        percentageLabel->setPositionY(percentageLabel->getPositionY() - 2.f);
+    }
+    if (nameColor == ccc3(233, 30, 99)) {
+        createBadge(createLayout("role-admin.png"_spr, 1.f, last_letter->convertToWorldSpace(getPosition()) + CCPoint{20.f, 6.f}, "globed-admin-badge", nullptr)).parent(this);
+        percentageLabel->setPosition({percentageLabel->getPositionX() + 16.f, percentageLabel->getPositionY() - 6.f});
+        percentageLabel->setPositionY(percentageLabel->getPositionY() - 2.f);
+    }
+    if (nameColor == ccc3(154, 88, 255)) {
+        createBadge(createLayout("role-supporter.png"_spr, 1.f, last_letter->convertToWorldSpace(getPosition()) + CCPoint{20.f, 6.f}, "globed-supporter-badge", nullptr)).parent(this);
+        percentageLabel->setPosition({percentageLabel->getPositionX() + 16.f, percentageLabel->getPositionY() - 6.f});
+        percentageLabel->setPositionY(percentageLabel->getPositionY() - 2.f);
+    }
+    if (nameColor == ccc3(248, 0, 255)) {
+        createBadge(createLayout("role-booster.png"_spr, 1.f, last_letter->convertToWorldSpace(getPosition()) + CCPoint{20.f, 6.f}, "globed-booster-badge", nullptr)).parent(this);
+        percentageLabel->setPosition({percentageLabel->getPositionX() + 16.f, percentageLabel->getPositionY() - 6.f});
+        percentageLabel->setPositionY(percentageLabel->getPositionY() - 2.f);
+    }
+
```

Comment:
once again, very ugly and improper use of layouts. they make your life a lot simpler, please use them. there is no need for most of this mess.

additionally, i don't really understand why this doesn't use the utility function and instead just copy pastes the code that checks for each name color